pkg/workflows/create.go

```diff
@@ -140,6 +137,9 @@ func (s *CreateBootStrapClusterTask) Name() string {
 // DeleteKindClusterTask implementation
 
 func (s *DeleteKindClusterTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	if commandContext.OriginalError != nil {
+		_ = s.CollectDiagnosticsTask.Run(ctx, commandContext)
```

Comment:
What about having a `CollectDiagnosticsTask` that has a next task field?
you could do `return &CollectDiagnosticsTask{nextTask: &DeleteKindClusterTask{})`

Reply:
ah yeah we could do that as well, I don't feel strongly either way